kernel/Cargo.toml

```diff
@@ -76,6 +77,7 @@ default-engine = [
   "parquet/object_store",
   "reqwest",
   "tokio",
+  "hdfs-native-object-store"
```

Comment:
This should go in the `cloud` feature I think. That's sort of the "catch-all" one for "i want to access 'cloud like' things.

Reply:
I've moved this under the "cloud" feature, including adding feature gates in `storage.rs` to allow compilation with the `cloud` feature enabled.